contracts/minter-imago/src/contract.rs

```diff
@@ -0,0 +1,666 @@
+
+    let recipient_addr = match recipient {
+        Some(some_recipient) => some_recipient,
+        None => info.sender.clone(),
+    };
+
+    let mint_price: Coin = mint_price(deps.as_ref(), is_admin)?;
+    // Exact payment only accepted
+    let payment = may_pay(&info, &config.unit_price.denom)?;
+    if payment != mint_price.amount {
+        return Err(ContractError::IncorrectPaymentAmount(
+            coin(payment.u128(), &config.unit_price.denom),
+            mint_price,
+        ));
+    }
+
+    let mut msgs: Vec<CosmosMsg<StargazeMsgWrapper>> = vec![];
+
+    // Create network fee msgs
+    let fee_percent = if is_admin {
+        Decimal::percent(AIRDROP_MINT_FEE_PERCENT as u64)
+    } else {
+        Decimal::percent(MINT_FEE_PERCENT as u64)
+    };
+    let network_fee = mint_price.amount * fee_percent;
+    let pw_fee = mint_price.amount * Decimal::percent(PW_MINT_FEE_PERCENT);
+    let addr = maybe_addr(deps.api, Some(DEV_ADDRESS.to_string()))?;
+    msgs.append(&mut pw_fee_msg(&info, pw_fee.u128(), addr.clone().unwrap())?);
+    msgs.append(&mut checked_fair_burn(&info, network_fee.u128(), addr.clone())?);
+
+    let mintable_tokens_result: StdResult<Vec<u32>> = MINTABLE_TOKEN_IDS
+        .keys(deps.storage, None, None, Order::Ascending)
+        .take(1)
+        .collect();
+    let mintable_tokens = mintable_tokens_result?;
+    if mintable_tokens.is_empty() {
+        return Err(ContractError::SoldOut {});
+    }
+    let mintable_token_id = mintable_tokens[0];
+
+    // Create mint msgs
+    let mint_msg = Cw721ExecuteMsg::Mint(MintMsg::<Empty> {
+        token_id: mintable_token_id.to_string(),
+        owner: recipient_addr.to_string(),
+        token_uri: Some(format!("{}/{}", config.base_token_uri, mintable_token_id)),
+        extension: Empty {},
+    });
+    let msg = CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: sg721_address.to_string(),
+        msg: to_binary(&mint_msg)?,
+        funds: vec![],
+    });
+    msgs.append(&mut vec![msg]);
+
+    // Remove mintable token id from map
+    MINTABLE_TOKEN_IDS.remove(deps.storage, mintable_token_id);
+    let mintable_num_tokens = MINTABLE_NUM_TOKENS.load(deps.storage)?;
+    // Decrement mintable num tokens
+    MINTABLE_NUM_TOKENS.save(deps.storage, &(mintable_num_tokens - 1))?;
+    // Save the new mint count for the sender's address
+    let new_mint_count = mint_count(deps.as_ref(), &info)? + 1;
+    MINTER_ADDRS.save(deps.storage, info.clone().sender, &new_mint_count)?;
+
+    // only allow one mint per address per block
+    let last_block_minter_sender = last_block_minted(deps.as_ref(), &info)?;
+    if last_block_minter_sender == env.block.height {
+        return Err(ContractError::MaxOneMintPerBlock {});
+    }
+    MINTER_LAST_BLOCK.save(deps.storage, info.clone().sender, &env.block.height)?;
+
+    let seller_amount = if !is_admin {
+        let amount = mint_price.amount - network_fee - pw_fee;
```

Comment:
I recently added this block to forward remaining amount to the admin. I probably didn't get this amount calculation correct. But the tests do not report any useful error. Any tips on how to debug this?

Reply:
this looks redundant to me, since `fee_percentage` is `AIRDROP_MINT_FEE_PERCENT`, which is 100% for admins.

Reply:
If i remove this block, then mint proceeds stay on the minter contract balance. I want the mint proceeds to all be forwarded to the admin

Reply:
Yep you were right about the airdrop 100% fee conflicting here. I fixed this now.